ai_sdlc/services/context7_service.py

```
@@ -29,27 +28,25 @@ def __init__(self, cache_dir: Path):
-        with open(self.cache_lock_file, 'r+') as lock_file:
-            while True:
-                try:
-                    fcntl.flock(lock_file.fileno(), fcntl.LOCK_EX | fcntl.LOCK_NB)
-                    return
-                except OSError as e:
-                    if time.time() - start_time > timeout:
-                        raise TimeoutError("Could not acquire cache lock") from e
-                    time.sleep(min(backoff, 1.0))  # Cap at 1 second
-                    backoff *= 2  # Exponential backoff
+        try:
+            self._lock_handle = portalocker.Lock(
```

Pylint (3.3.7)

[warning] Attribute '_lock_handle' defined outside __init__

(W0201)